### PR TITLE
Added NTLM Authentication based on presence of NTLM Domain property

### PR DESCRIPTION
**Pull request by peetkes**
Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

This pullrequests adds NTLM Authentication support to the InvokeHTTP processor so it can communicate to NTLM authenticated webservices like Sharepoint

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?
No

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
No, there is no JIRA Ticket for NTLM Authentication

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
Yes

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._
Yes

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on both JDK 8 and JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [x] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java [hunk 1]
```diff
@@ -113,9 +114,10 @@
 import static org.apache.commons.lang3.StringUtils.trimToEmpty;
 
 @SupportsBatching
-@Tags({"http", "https", "rest", "client"})
+@Tags({"http", "https", "rest", "client","ntlm"})
 @InputRequirement(Requirement.INPUT_ALLOWED)
 @CapabilityDescription("An HTTP client processor which can interact with a configurable HTTP Endpoint. The destination URL and HTTP Method are configurable."
+    + " Also supports NTLM authentication to communicate with a sharepoint webservice"
     + " FlowFile attributes are converted to HTTP headers and the FlowFile contents are included as the body of the request (if the HTTP Method is PUT, POST or PATCH).")
```
Comment (ottobackwards) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java]: Unless NTLM _only_ works with sharepoint, there isn't a need to specify it here.  If you _do_ specify it, you might want to capitalize it correctly.

Comment (peetkes) [reply nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/InvokeHTTP.java]: Adjusted the Description, works with NTLM in general

**Comment by peetkes** [comment]
Failed tests in modules I did not touch....

**Comment by joewitt** [comment]
....this looks like a really great addition.  However, it is clearly largely code copied from stackoverflow/etc..  We cannot take such things unless the copyright/licensing is clear.  Ideally this comes from some library for which license/attribution is clear.

**Comment by peetkes** [comment]
So what would you need me to do to get this accepted?

**Comment by mdjaniga** [comment]
believe I found a JIRA ticket for this: NIFI-3096

**Comment by joewitt** [comment]
The code needs to be originally authored OR it needs to be from a source that we can properly account for in terms of its licensing and copyright.

Otherwise we simply cannot accept it.  Dont get me wrong - this is great and looks like it would really help folks.  But we need originally authored code, or code that is from a category A license group (https://www.apache.org/legal/resolved.html#category-a), or binary dependencies that come from either category A or category B (https://www.apache.org/legal/resolved.html#category-b)

I hope this helps a little more.

**Comment by CDann3r** [comment]
I think the issue is that the used httpClient (okhttp3.OkHttpClient) doesn't support NTLM. Therefore two complete java classes have been added which handle the actual NTLM.

I propose to use "org.apache.http.impl.client.HttpClientBuilder" as @simonellistonball did with his WebDAV Processor which is Licensed under Apache2.

Reference => see addAuth method which creates a CredentialProvider for the HttpClientBuilder
[AbstractWebDAVProcessor.java](https://github.com/simonellistonball/nifi-webdav-bundle/blob/master/nifi-webdav-processors/src/main/java/org/apache/nifi/processors/webdav/AbstractWebDAVProcessor.java)

**Comment by dazhitchman** [comment]
Any progress on this ? also noting NIFI-3096 refers to https://github.com/simonellistonball/nifi-webdav-bundle (Apache 2 license) that could be leveraged to add functionality.

**Pull request closed without merging**